src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,125 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+const baseFormData = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [newMovie, setNewMovie] = useState(baseFormData);
+  const {
+    title, description, imgUrl, imdbUrl, imdbId,
+  } = newMovie;
+
+  const [count, setCount] = useState(0);
+
+  const isAnyFieldEmpty
+    = !title.trim()
+    || !imgUrl.trim()
+    || !imdbUrl.trim()
+    || !imdbId.trim();
+
+  const handleMovieFieldChange = (name: string, value: string) => {
+    setNewMovie((prevMovie) => ({ ...prevMovie, [name]: value }));
+  };
+
+  function isValidUrl(url: string) {
+    if (!url) {
+      return undefined;
```

Comment:
```suggestion
      return;
```
empty return by default returns undefined